src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt

```diff
@@ -135,6 +138,23 @@ class CodyToolWindowContent(private val project: Project) {
     allContentLayout.show(allContentPanel, MAIN_PANEL)
   }
 
+  private fun selectHistory(state: ChatState) {
+    val session =
+        AgentChatSession.getSessionByInternalId(state.internalId!!)
```

Comment:
I very much do not like those `!!`.
If we are so certain it cannot be null it should be reflected in the types.

Reply:
Something being `null` when we do not expect it should be rare case, no? I mean, some serialisation error or whatever?

If that is the case, then I suggest we validate the model when we load if from disk, and if it's incoherent in any way we simply delete that session. There is no point in showing users chats which they can't open anyway.
This way you can be sure there are no nulls in the model later and interfaces can be clear.

Reply:
I removed the null checks from the code and slightly rewrote the state to be more defensive.

Ultimately, I would like the state to be completely hidden outside of HistoryService however this required more work. I have reviewed other implementations and it seems that the problem is not solved at all (and nullable leaks into the logic) or a persistent state component hides the state from the world, providing custom setters/getters on component layer. 

Example: https://github.com/JetBrains/intellij-community/blob/master/plugins/git4idea/src/git4idea/config/GitVcsSettings.java